### PR TITLE
15.0 mig purchase request department

### PR DESCRIPTION
**Pull request by JoelZilli**

### CODE REVIEW

**Review by ajaniszewska-dev — APPROVED** [review]

File: purchase_request_department/models/purchase_request.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright 2017-2020 Forgeflow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class PurchaseRequest(models.Model):
+    _inherit = "purchase.request"
+
+    def _get_my_department(self):
+        employees = self.env.user.employee_ids
```
Comment (sswapnesh) [inline purchase_request_department/models/purchase_request.py:10]: we should use `employee_id` field from user

https://github.com/odoo/odoo/blob/b49a23fdd83c139f19c22017667005de2dd8cc75/addons/hr/models/res_users.py#L83

File: purchase_request_department/models/purchase_request.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        return (
+            employees[0].department_id
```
Comment (sswapnesh) [inline purchase_request_department/models/purchase_request.py:12]: Same  for`department_id`  https://github.com/odoo/odoo/blob/b49a23fdd83c139f19c22017667005de2dd8cc75/addons/hr/models/res_users.py#L92

File: purchase_request_department/models/purchase_request.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+class PurchaseRequestLine(models.Model):
+    _inherit = "purchase.request.line"
+
+    department_id = fields.Many2one(
+        comodel_name="hr.department",
+        related="request_id.department_id",
+        store=True,
+        string="Department",
+        readonly=True,
```
Comment (sswapnesh) [inline purchase_request_department/models/purchase_request.py:37]: related fields are readonly by default

**Review by sswapnesh — CHANGES_REQUESTED** [review]

**Comment by JoelZilli** [comment]
@sswapnesh 
All changes done!

**Review by TheerayutEncoder — APPROVED** [review]
LGTM!!! :+1:

**Review by sswapnesh — APPROVED** [review]

**Pull request merged**